src/rabbitalerter.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * An alerting system for message queues.
+ * @module rabbitalerter
+ */
+const assert = require('assert');
+const taskcluster = require('taskcluster-client');
+
+/**
+ * An interface used to generate various TaskCluster Notifications.
+ *
+ * @class RabbitAlerter
+ */
+class RabbitAlerter {
+  /**
+   * @param {number} config.alerter.messageCountTolerance       - When a queue holds more messages
+   *                                                              than this tolerance, an alert will be created.
+   * @param {number} config.alerter.messagePublishRateTolerance - When the number of messages published per second
```

Comment:
Sorry for bumping this up so late. I was struggling in the swamps of deadlines and exams last week...

@jonasfj 's idea of inferring max age (oldest messages) from `message_stats.publish` and `message_stats.ack` sounds awesome to me, provided that these two counters are reliable (and are what we think they are - monotonic counters of the whole history since the server starts). In the [latest doc of rabbitmq-management](https://raw.githack.com/rabbitmq/rabbitmq-management/rabbitmq_v3_6_6/priv/www/doc/stats.html), `publish` is " Count of messages published. ", which doesn't really help much. Besides, there is no `ack` in the latest doc. It might have been renamed in recent versions, or is undocumented. I do notice that rabbitmq introduced breaking changes to the management API sometimes, and we need to check the doc of the version on the production servers to be sure. (As a side note, it is hence somewhat hard to make a reusable package for the rabbitmq management API because of the version dependency.)

FWIW, I just checked out how [Sensu](https://sensuapp.org/), a monitoring system I used before, monitors rabbitmq. One metric they chose is the (estimated) "draining time":
`messages` / `backing_queue_status.avg_egress_rate` ([code](https://github.com/sensu-plugins/sensu-plugins-rabbitmq/blob/master/bin/check-rabbitmq-queue-drain-time.rb)).

Again, it is not clear to me in what time period the rate is measured and how it is averaged (I'm guessing it measures rates at a short sampling interval and averages the rates over a few intervals). If the `avg_egress_rate` is averaged over a long period, we might give false alerts after "processable" spikes (a spike in a low-traffic queue but consumers are able to crank up).

Anyway, this is a good starting point as we have the basic facilities for monitoring and alerting now. Nice work @itsbriany !